File: VRTInterstitialCustomEventVungle.m
//
//  VRTInterstitialCustomEventGoogleMobileAds.m
//
//  Created by Scott McCoy on 5/9/19.
//  Copyright © 2019 VRTCAL. All rights reserved.
//

//Header
#import "VRTInterstitialCustomEventVungle.h"

//Dependencies
#import "VRTVungleManager.h"


@interface VRTInterstitialCustomEventVungle()
@property NSString *placementId;
@end

@implementation VRTInterstitialCustomEventVungle

- (void) loadInterstitialAd {
    self.placementId = [self.customEventConfig.thirdPartyCustomEventData objectForKey:@"adUnitId"];
    
    if (self.placementId == nil) {
        VRTError *error = [VRTError errorWithCode:VRTErrorCodeCustomEvent message:@"No placement Id"];
        [self.customEventLoadDelegate customEventFailedToLoadWithError:error];
        return;
    }
    
    NSError *error = [[VRTVungleManager singleton]
        loadPlacementWithID:self.placementId
        withSize:VungleAdSizeUnknown
        vrtVungleManagerDelegate:(id <VRTVungleManagerDelegate>) self
    ];
    
    if (error) {
        [self.customEventLoadDelegate customEventFailedToLoadWithError:error];
        return;
    }
}

- (void) showInterstitialAd {
    UIViewController *vc = [self.viewControllerDelegate vrtViewControllerForModalPresentation];
    [[VRTVungleManager singleton] showInterstitial:self.placementId viewController:vc];
}


#pragma mark - VRTVungleManagerDelegate
- (void)vungleAdViewedForPlacement:(nullable NSString *)placementID {
    [self.customEventShowDelegate customEventShown];
}

- (void)vungleDidCloseAdForPlacementID:(nonnull NSString *)placementID {
    [self.customEventShowDelegate customEventDidDismissModal:VRTModalTypeUnknown];
}

- (void)vungleDidShowAdForPlacementID:(nullable NSString *)placementID {
    [self.customEventShowDelegate customEventShown];
}

- (void)vungleRewardUserForPlacementID:(nullable NSString *)placementID {
    // No VRT analog
}

- (void)vungleTrackClickForPlacementID:(nullable NSString *)placementID {
    [self.customEventShowDelegate customEventClicked];
}

- (void)vungleWillCloseAdForPlacementID:(nonnull NSString *)placementID {
    [self.customEventShowDelegate customEventWillDismissModal:VRTModalTypeUnknown];
}

 - (void)vungleWillLeaveApplicationForPlacementID:(nullable NSString *)placementID {
    [self.customEventShowDelegate customEventWillLeaveApplication];
}

- (void)vungleWillShowAdForPlacementID:(nullable NSString *)placementID {
    // No VRT Analog
}

- (void)vungleAdPlayabilityUpdate:(BOOL)isAdPlayable placementID:(nullable NSString *)placementID error:(nullable NSError *)error {
    
    if (error) {
        [self.customEventLoadDelegate customEventFailedToLoadWithError:error];
        return;
    }
    
    if (!isAdPlayable) {
        VRTError *vrtError = [VRTError errorWithCode:VRTErrorCodeNoFill message:@"Vungle Ad Not Playable"];
        [self.customEventLoadDelegate customEventFailedToLoadWithError:vrtError];
        return;
    }
    
    [self.customEventLoadDelegate customEventLoaded];
}
@end

File: VRTInterstitialCustomEventVungle.swift
//  Converted to Swift 5.8.1 by Swiftify v5.8.26605 - https://swiftify.com/
//
//  VRTBannerCustomEventGoogleMobileAds.h
//
//  Created by Scott McCoy on 5/9/19.
//  Copyright © 2019 VRTCAL. All rights reserved.
//

//
//  VRTInterstitialCustomEventGoogleMobileAds.m
//
//  Created by Scott McCoy on 5/9/19.
//  Copyright © 2019 VRTCAL. All rights reserved.
//

//Header
//Vungle Banner Adapter, Vrtcal as Primary

class VRTInterstitialCustomEventVungle: VRTAbstractInterstitialCustomEvent, VRTVungleManagerDelegate {
    private var placementId: String?

    func loadInterstitialAd() {
        placementId = customEventConfig.thirdPartyCustomEventData["adUnitId"] as? String

        if placementId == nil {
            let error = VRTError(code: VRTErrorCodeCustomEvent, message: "No placement Id")
            customEventLoadDelegate.customEventFailedToLoadWithError(error)
            return
        }

        let error = VRTVungleManager.singleton().loadPlacement(
            withID: placementId ?? "",
            with: VungleAdSizeUnknown,
            vrtVungleManagerDelegate: self as? VRTVungleManagerDelegate)

        if let error {
            customEventLoadDelegate.customEventFailedToLoadWithError(error)
            return
        }
    }

    func showInterstitialAd() {
        let vc = viewControllerDelegate.vrtViewControllerForModalPresentation()
        VRTVungleManager.singleton().showInterstitial(placementId ?? "", viewController: vc)
    }

    // MARK: - VRTVungleManagerDelegate

    func vungleAdViewed(forPlacement placementID: String?) {
        customEventShowDelegate.customEventShown()
    }

    func vungleDidCloseAd(forPlacementID placementID: String) {
        customEventShowDelegate.customEventDidDismissModal(VRTModalTypeUnknown)
    }

    func vungleDidShowAd(forPlacementID placementID: String?) {
        customEventShowDelegate.customEventShown()
    }

    func vungleRewardUser(forPlacementID placementID: String?) {
        // No VRT analog
    }

    func vungleTrackClick(forPlacementID placementID: String?) {
        customEventShowDelegate.customEventClicked()
    }

    func vungleWillCloseAd(forPlacementID placementID: String) {
        customEventShowDelegate.customEventWillDismissModal(VRTModalTypeUnknown)
    }

    func vungleWillLeaveApplication(forPlacementID placementID: String?) {
        customEventShowDelegate.customEventWillLeaveApplication()
    }

    func vungleWillShowAd(forPlacementID placementID: String?) {
        // No VRT Analog
    }

    func vungleAdPlayabilityUpdate(_ isAdPlayable: Bool, placementID: String?, error: Error?) {

        if let error {
            customEventLoadDelegate.customEventFailedToLoadWithError(error)
            return
        }

        if !isAdPlayable {
            let vrtError = VRTError(code: VRTErrorCodeNoFill, message: "Vungle Ad Not Playable")
            customEventLoadDelegate.customEventFailedToLoadWithError(vrtError)
            return
        }

        customEventLoadDelegate.customEventLoaded()
    }
}

//Dependencies